nix/status-go/shared/build.nix

```diff
@@ -0,0 +1,141 @@
+{ stdenv, lib, fetchFromGitHub, buildGo114Package
+# Dependencies
+, xcodeWrapper
+, go, androidPkgs
+# metadata and status-go source
+, meta, source
+# build parameters
+, platform ? "android"
+, arch ? "386"
+, api ? "23" }:
+
+let
+  inherit (lib) attrNames getAttr strings concatStringsSep concatMapStrings;
+
+  removeReferences = [ go ];
+  removeExpr = refs: ''remove-references-to ${concatMapStrings (ref: " -t ${ref}") refs}'';
+
+  targetArchMap = rec {
+    "386" = "i686";
+    "arm" = "armv7a";
+    "arm64" = "aarch64";
+  };
+  ldArchMap = {
+    "386" = "x86";
+    "arm" = "arm";
+    "arm64" = "arm64";
+  };
+
+  # Shorthands for the built phase
+  targetArch = getAttr arch targetArchMap;
+  ldArch = getAttr arch ldArchMap;
+
+  osId = builtins.elemAt (builtins.split "\-" stdenv.hostPlatform.system) 2;
+  osArch = builtins.elemAt (builtins.split "\-" stdenv.hostPlatform.system) 0;
+
+  androidTarget = targetArch + "-linux-" + platform;
+
+  iosArchMap = {
+    "386" = "x86_64";
+    "arm" = "armv7";
+    "arm64" = "arm64";
+  };
+  iosSdk = if arch == "386" then "iphonesimulator" else "iphoneos";
+  iosArch = getAttr arch iosArchMap;
+  compilerFlags = if platform == "android" || platform == "androideabi" then
+  "\"-isysroot $ANDROID_NDK_HOME/toolchains/llvm/prebuilt/${osId}-${osArch}/sysroot -target ${androidTarget}${api}\""
+  else
+  "\"-isysroot $(xcrun --sdk ${iosSdk} --show-sdk-path) -miphonesimulator-version-min=7.0 -fembed-bitcode -arch ${iosArch}\"";
```

Comment:
![1376072970686](https://user-images.githubusercontent.com/2212681/94688722-42430700-032e-11eb-8bbf-021d951d303d.jpg)
There has to be a cleaner way of doing this. I'll look into it.